### PR TITLE
ZPS-5997: add u/tests for tags and models

### PR DESCRIPTION
**Pull request by alexvashchyshyn**
[ZPS-5997](https://jira.zenoss.com/browse/ZPS-5997)

### CODE REVIEW

File: pkg/backends/zenoss/tags_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	t.Run("comparing values", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		}))
+		defer server.Close()
+
+		client, err := NewClient(server.URL, "apiKey123", false, false, 2, 2, 2, []string{"test:", "go"}, []string{"go", "lang"}, []string{"go", "stats"}, []string{"test:test_tag", "go"}, 1, float64(0.2), []string{"a", "b"}, 1*time.Second, gostatsd.TimerSubtypes{})
+		request1 := client.getTags([]string{"go:", "go"})
+		request2 := client.getTags([]string{"stats:", "abc"})
+		request3 := client.getTags([]string{"failed", "request"})
```
Comment (joshw) [inline pkg/backends/zenoss/tags_test.go]: It looks like you cover some of the variants in TestParseTag, bug still, i think these 3 test requests could be a bit more specific.

Maybe one with a tag name and value, one with a tag name ('true' value) and an invalid tag name, and one with nothing?

**Review by joshw — CHANGES_REQUESTED** [review]

File: pkg/backends/zenoss/tags_test.go (hunk 1 above)
Comment (alexvashchyshyn) [reply pkg/backends/zenoss/tags_test.go]: added 4 specific requests: one with a tag name and value, one with a tag name ('true' value) and an invalid tag name, and one with nothing

File: pkg/backends/zenoss/tags_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			result := parseTag(test.input)
+			assert.EqualValues(t, test.expected, *result)
+		}
+	})
+
+}
+
+func TestValueFromString(t *testing.T) {
+
+	t.Run("cheking not empty", func(t *testing.T) {
```
Comment (joshw) [inline pkg/backends/zenoss/tags_test.go]:
```suggestion
	t.Run("checking not empty", func(t *testing.T) {
```

File: pkg/backends/zenoss/tags_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+			result := valueFromString(test.input)
+			assert.EqualValues(t, test.expected, *result)
+		}
+	})
+
+}
+
+func TestValueFromStringSlice(t *testing.T) {
+
+	t.Run("cheking not empty", func(t *testing.T) {
```
Comment (joshw) [inline pkg/backends/zenoss/tags_test.go]:
```suggestion
	t.Run("checking not empty", func(t *testing.T) {
```

**Review by joshw — APPROVED** [review]
Approved, though there is a minor spelling error to be corrected

**Pull request merged**